mozfldp/server.py

```diff
@@ -26,53 +31,69 @@ class ServerFacade:
     into a classifier.
     """
 
-    def __init__(self):
-        self._classifier = SGDClassifier(loss="hinge", penalty="l2")
-
-        self._buffer = {"some_client_uuid_1": [], "some_client_uuid_2": []}
-
-    def ingest_client_data(self, client_json):
-        # TODO: we need to read one sample of data from a client and
-        # add it to a sample buffer.  The intent here is that we want
-        # to collect enough data that we can safely sample from the
-        # buffer
-
-        # Assume that client_json is some JSON blob with everything we
-        # need to call client_update successfully.  We should have a
-        # `client_id` key that uniquely identifies the client so that
-        # we can pin data to a particular client.
-
-        # TODO: maybe something like this?
-        # self._buffer[client_json['client_id']].append(client_json)
-        pass
-
-    def update_classifier(self):
-        # TODO: this is where the magic happens
-        #
-        # I think roughly, this should be applying a sample of client
-        # data through a training loop.
-        #
-        # I believe you want to take the code from:
-        # https://github.com/mozilla/CANOSP-2019/blob/master/simulation_util.py#L98
-        # to
-        # https://github.com/mozilla/CANOSP-2019/blob/master/simulation_util.py#L149
-        # and update the parameters of the SGDClassifier.
-        #
-        # The samples `S` in simulation_util.py would be equivalent to self._buffer
-        # in this class
-        pass
-
-    def classify(self, some_input_json):
+    def __init__(self, coef, intercept, num_client, client_fraction):
         """
-        Apply the classifer to some sample input
+        TOOD: add detailed docstring explaining each argument,
+        referencing the paper
         """
-        reshaped_X_test = None
-        reshaped_Y_test = None
-        score = self._classifier.score(reshaped_X_test, reshaped_Y_test)
-        return score
+        self._coef = coef
+        self._intercept = intercept
+
+        self._num_client = num_client
+        self._client_fraction = client_fraction
+        # grab all the weights from clients
+        self._client_coefs = None
+        self._client_intercepts = None
+        self._num_samples = []
+
+    def send_weights(self, coefs, intercept, num_features):
```

Comment:
IIUC something like this would get called from `client_update` below to handle incoming payloads from clients. I think we want to use an approach like `ingest_client_data` from the original version instead.

Reply:
That makes sense. I reverted to the old `ingest_client_data` method and pass the json payload instead.